addons/api/addon/serializers/application.js

```diff
@@ -88,6 +88,13 @@ export default class ApplicationSerializer extends RESTSerializer {
         delete json[key];
       }
     }
+    if (
+      options.trimWhitespace &&
+      json.attributes[attribute?.name] &&
```

Comment:
I know this is still a WIP but wanted to point out that this makes the assumption that the field is _only_ an attribute that needs to be trimmed but it could also be a field that is top level as well, such as `name`. We should aim to be able to handle it for any field on the model.